### PR TITLE
ROX-17365: Delegated Scanning Kill Switch

### PR DESCRIPTION
**Pull request by dcaravel**
## Description

Adds an environment variable that disables the capabilities and secret storage associated with delegated image scanning.

## Checklist
- [x] Investigated and inspected CI test results
- [x] Unit test and regression tests added
- [x] Evaluated and added CHANGELOG entry if required
- [x] Determined and documented upgrade steps
- [x] ~Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

If any of these don't apply, please comment below.

## Testing Performed
Unit testing, local deployment. 

Verified:

- Central delegated registry API does not show the cluster as a valid delegation target
- Verified logs show additional registries are not being upserted (ie: from global pull secret)
- Scans of OCP internal registry continue to work as expected
- Ad-hoc requests for image scanning to the cluster encounter error as expected

#### Central delegated registry API does not show the cluster as a valid delegation target
```
❯ k get deploy/sensor -o yaml | grep -A 40 "env"
        env:
        ...
        - name: ROX_SCANNER_GRPC_ENDPOINT
          value: scanner.stackrox.svc:8443
        - name: ROX_LOCAL_IMAGE_SCANNING_ENABLED
          value: "true"
        - name: ROX_DELEGATED_SCANNING_DISABLED
          value: "true"

❯ curl ... https://<ip>/v1/delegatedregistryconfig/clusters
{"clusters":[{"id":"44277e32-dfc2-4754-838d-2fea0e0191c4","name":"local-scan","isValid":false}]}
```

### Verified logs show additional registries are not being upserted (ie: from global pull secret)
With kill switch
```sh
Debug: Upserted registry "172.30.162.191:5000" for namespace "openshift-apiserver" into store
Debug: Upserted registry "172.30.162.191:5000" for namespace "openshift-cloud-controller-manager-operator" into store
Debug: Upserted registry "image-registry.openshift-image-registry.svc.cluster.local:5000" for namespace "openshift-apiserver" into store
Debug: Upserted registry "image-registry.openshift-image-registry.svc.cluster.local:5000" for namespace "openshift-cloud-controller-manager-operator" into store
Debug: Upserted registry "image-registry.openshift-image-registry.svc:5000" for namespace "openshift-apiserver" into store
Debug: Upserted registry "image-registry.openshift-image-registry.svc:5000" for namespace "openshift-cloud-controller-manager-operator" into store
```
Without kill switch:
```sh
Debug: Upserted global registry "quay.io" into store
Debug: Upserted global registry "registry.connect.redhat.com" into store
...
Debug: Upserted registry "172.30.162.191:5000" for namespace "openshift-apiserver" into store
Debug: Upserted registry "172.30.162.191:5000" for namespace "openshift-cloud-controller-manager-operator" into store
Debug: Upserted registry "image-registry.openshift-image-registry.svc.cluster.local:5000" for namespace "openshift-apiserver" into store
Debug: Upserted registry "image-registry.openshift-image-registry.svc.cluster.local:5000" for namespace "openshift-cloud-controller-manager-operator" into store
Debug: Upserted registry "image-registry.openshift-image-registry.svc:5000" for namespace "openshift-apiserver" into store
Debug: Upserted registry "image-registry.openshift-image-registry.svc:5000" for namespace "openshift-cloud-controller-manager-operator" into store
```

### Scans of OCP internal registry continue to work as expected

**Sensor logs**
```
...
pkg/logging: 2023/06/03 19:38:01.788523 logger.go:186: Debug: Sending scan to central for image "quay.io/rhacs-eng/scanner-db-slim:4.0.0"
pkg/logging: 2023/06/03 19:38:01.789383 logger.go:186: Debug: Sending scan to local scanner for image "image-registry.openshift-image-registry.svc:5000/default/nginx:latest"
pkg/logging: 2023/06/03 19:38:01.791311 logger.go:186: Debug: Sending scan to central for image "quay.io/openshift-release-dev/ocp-release@sha256:74b23ed4bbb593195a721373ed6693687a9b444c97065ce8ac653ba464375711"
...
```
image

### Ad-hoc requests for image scanning to the cluster encounter error as expected
```sh
❯ curl https://<ip>/v1/delegatedregistryconfig/clusters ...
{"clusters":[{"id":"44277e32-dfc2-4754-838d-2fea0e0191c4","name":"local-scan","isValid":false}]}

❯ roxctl image scan --image=hello-world
ERROR:	Scanning image failed: could not scan image: "hello-world": rpc error: code = Internal desc = cluster "44277e32-dfc2-4754-838d-2fea0e0191c4" does not support delegated scanning. Retrying after 3 seconds...
```

## PR Stack
- **This PR** - https://github.com/stackrox/stackrox/pull/6312
- https://github.com/stackrox/stackrox/pull/6277
- https://github.com/stackrox/stackrox/pull/6229
- https://github.com/stackrox/stackrox/pull/6186
- https://github.com/stackrox/stackrox/pull/6005
- https://github.com/stackrox/stackrox/pull/5951

### CODE REVIEW

**Review by connorgorman — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -19,6 +19,7 @@ Please avoid adding duplicate information across this changelog and JIRA/doc inp
   Note that if the environment variable is set, no signatures will be fetched and thus the signature verification feature cannot be used.
 - ROX-16532: Resource limits and requests for the node-inventory container can now be configured via the operator.
 - A new environment variable `ROX_SCAN_TIMEOUT` has been added to Sensor which allows for customizing the image scan timeout used in Sensor initiated scans.
+- ROX-17365: A new environment variable `ROX_DELEGATED_SCANNING_DISABLED` has been added that disables delegated scanning capabilities while leaving other local scanning capabilities in-tact.
```
Comment (RTann) [inline CHANGELOG.md:22]: nit: `in-tact` -> `intact`

Also, what are other local scanning capabilities?

File: sensor/kubernetes/listener/resources/secrets.go [hunk 2]
```diff
@@ -289,7 +289,7 @@ func (s *secretDispatcher) processDockerConfigEvent(secret, oldSecret *v1.Secret
 				}
 			}
 
-			if env.LocalImageScanningEnabled.BooleanSetting() {
+			if env.LocalImageScanningEnabled.BooleanSetting() && !env.DelegatedScanningDisabled.BooleanSetting() {
```
Comment (RTann) [inline sensor/kubernetes/listener/resources/secrets.go:292]: can you add a comment above this to make it clear anything within this closure is related to delegated scanning?

**Review by RTann — APPROVED** [review]
approving, but two tiny nits

File: CHANGELOG.md (hunk 1 above)
Comment (dcaravel) [reply CHANGELOG.md:22]: Scanning of images from the OCP internal registry

**Pull request merged**